### ISSUE TITLE
Make UnzipTransform public API

### ISSUE THREAD

**Issue by liutikas** [gradle/gradle#21443]
Make `org.gradle.api.internal.artifacts.transform.UnzipTransform` public API. It is a generally useful TransformAction (e.g. [usage by Paparazzi plugin](https://github.com/cashapp/paparazzi/blob/master/paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt#L224) as it just end up being copied by dozens of plugins https://cs.github.com/?scopeName=All+repos&scope=&q=TransformAction+UnzipTransform

### Expected Behavior
`org.gradle.api.internal.artifacts.transform.UnzipTransform` is public API.

### Current Behavior
`org.gradle.api.internal.artifacts.transform.UnzipTransform` is internal API.

### Context
See usages and copies above.

**Comment by reinsch82** [comment]
This feature request is in the backlog of the relevant team and is prioritized by them.

**Comment by hfhbd** [comment]
Is there any reason to not use this code? `ArchiveOperations` also supports tar, and gzip + bzip2. 
```kotlin
@CacheableTransform
abstract class UnzipTransform : TransformAction<TransformParameters.None> {
    @get:InputArtifact
    @get:PathSensitive(PathSensitivity.NAME_ONLY)
    abstract val inputArtifact: Provider<FileSystemLocation>

    @get:Inject
    abstract val archiveOperations: ArchiveOperations

    @get:Inject
    abstract val fileSystemOperations: FileSystemOperations

    override fun transform(outputs: TransformOutputs) {
        val zipFile = inputArtifact.get().asFile
        val outputDir = outputs.dir(zipFile.nameWithoutExtension)

        fileSystemOperations.copy {
            it.from(archiveOperations.zipTree(zipFile))
            it.into(outputDir)
        }
    }
}
```

**Comment by liutikas** [comment]
Yes it can be done without this API but given how common of a use it can be it makes sense to have it built in rather than everyone creating their own version

**Comment by hfhbd** [comment]
@liutikas I was more surprised that the internal `UnzipTransform` uses another code path instead of `FileSystemOperations` and `ArchiveOperations`.
